### PR TITLE
Tdl 6173 bookmark key handling

### PR DESCRIPTION
**Pull request by dbshah1212**
# Description of change
- Bookmark key handling

# Manual QA steps
 - Run the sync mode with state file having bookmark_date without timezone and no error thrown. 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_chargebee/streams/base.py [hunk 1]
```diff
@@ -132,24 +130,26 @@ def sync_data(self):
         if bookmark_date is None:
             LOGGER.info('Could not locate bookmark_date from STATE file. Falling back to start_date from config.json instead.')
             bookmark_date = get_config_start_date(self.config)
+            try:
+                # Check start date format
+                singer.utils.strptime(self.config.get("start_date"))
+            except ValueError:
+                raise ValueError("start_date must be in '%Y-%m-%dT%H:%M:%SZ' format")
```
Comment (karanpanchal-crest) [inline tap_chargebee/streams/base.py:137]: Add exception error message in the above message

Comment (dbshah1212) [reply tap_chargebee/streams/base.py:137]: As discussed offline, this won't require.

**Review by karanpanchal-crest — APPROVED** [review]

**Review by KAllan357 — CHANGES_REQUESTED** [review]
Can you explain why the bookmarking logic is being reverted back? The tests are also broken here.

**Comment by dbshah1212** [comment]
Hi @KAllan357, the Chargebee team implemented the wrong logic so instead of storing the last record's time they were storing `now-2 min` time. Hence I reverted that logic as we discussed. 
Ref: https://talend.slack.com/archives/C01TGC4QAUE/p1624368808004000
Test where broke because a Chargebee product catalog version 2 account was expired. Tests are passing now.

**Review by KAllan357 — APPROVED** [review]

**Comment by cb-nandita** [comment]
> Hi @KAllan357, the Chargebee team implemented the wrong logic so instead of storing the last record's time they were storing `now-2 min` time. Hence I reverted that logic as we discussed. Ref: https://talend.slack.com/archives/C01TGC4QAUE/p1624368808004000 Test where broke because a Chargebee product catalog version 2 account was expired. Tests are passing now.

@KAllan357 We implemented this logic of storing 'now-2 min', in order to make sure, none of the records get missed, during the API calls if updates had happened concurently. Please refer [doc](https://apidocs.chargebee.com/docs/api#pagination_and_filtering). 
We actually made these changes a fix as many customers' had reported missing records issue. Could you please revert it back as I see customers coming back now with the same issue ?

**Comment by savan-chovatiya** [comment]
> > Hi @KAllan357, the Chargebee team implemented the wrong logic so instead of storing the last record's time they were storing `now-2 min` time. Hence I reverted that logic as we discussed. Ref: https://talend.slack.com/archives/C01TGC4QAUE/p1624368808004000 Test where broke because a Chargebee product catalog version 2 account was expired. Tests are passing now.
> 
> @KAllan357 @[dbshah1212](https://github.com/dbshah1212) implemented this logic of storing 'now-2 min', in order to make sure, none of the records get missed, during the API calls if updates had happened concurently. Please refer [doc](https://apidocs.chargebee.com/docs/api#pagination_and_filtering). We actually made these changes a fix as many customers' had reported missing records issue. Could you please revert it back as I see customers coming back now with the same issue ? We could also get into a call and discuss more on this. Please let me know if any other details are required.

@cb-nandita, We had updated the bookmark mechanism previously because if the server has some load and data is coming 3-5 minutes late then we might miss data because the bookmark was written with now-2min.  We kept the maximum replication_key value from all the records as a bookmark because data was coming in order. 

We have created a new #86  with the below bookmarking mechanism which will ensure to keep bookmark at max now-2min. 
`bookmark = min(max_replication_key, now-2min)`

Can you please review the above PR once? let us know if you have any questions.

**Pull request merged**